consensus/hotstuff/pacemaker/pacemaker_test.go

```diff
@@ -30,456 +32,188 @@ func expectedTimerInfo(view uint64, mode model.TimeoutMode) interface{} {
-	nve, nveOccurred := pm.ProcessQC(qc)
-	notifier.AssertExpectations(t)
-	assert.Equal(t, uint64(4), pm.CurView())
-	assert.True(t, nveOccurred && nve.View == 4)
+	s.persist.On("PutLivenessData", LivenessData(qc)).Return(nil).Once()
+	s.notifier.On("OnStartingTimeout", expectedTimerInfo(4, model.ReplicaTimeout)).Return().Once()
+	s.notifier.On("OnQcTriggeredViewChange", qc, uint64(4)).Return().Once()
+	nve, err := s.paceMaker.ProcessQC(qc)
+	require.NoError(s.T(), err)
+	s.notifier.AssertExpectations(s.T())
+	require.Equal(s.T(), uint64(4), s.paceMaker.CurView())
+	require.True(s.T(), nve.View == 4)
+	require.Equal(s.T(), qc, s.paceMaker.NewestQC())
+	require.Nil(s.T(), s.paceMaker.LastViewTC())
 
 	qc = QC(12)
-	notifier.On("OnStartingTimeout", expectedTimerInfo(13, model.ReplicaTimeout)).Return().Once()
-	notifier.On("OnQcTriggeredViewChange", qc, uint64(13)).Return().Once()
-	nve, nveOccurred = pm.ProcessQC(qc)
-	assert.True(t, nveOccurred && nve.View == 13)
-
-	notifier.AssertExpectations(t)
-	assert.Equal(t, uint64(13), pm.CurView())
+	s.persist.On("PutLivenessData", LivenessData(qc)).Return(nil).Once()
+	s.notifier.On("OnStartingTimeout", expectedTimerInfo(13, model.ReplicaTimeout)).Return().Once()
+	s.notifier.On("OnQcTriggeredViewChange", qc, uint64(13)).Return().Once()
+	nve, err = s.paceMaker.ProcessQC(qc)
+	require.NoError(s.T(), err)
+	require.True(s.T(), nve.View == 13)
+	require.Equal(s.T(), qc, s.paceMaker.NewestQC())
+	require.Nil(s.T(), s.paceMaker.LastViewTC())
+
+	s.notifier.AssertExpectations(s.T())
+	require.Equal(s.T(), uint64(13), s.paceMaker.CurView())
 }
 
-// Test_IgnoreOldBlocks tests that PaceMaker ignores old blocks
-func Test_IgnoreOldQC(t *testing.T) {
-	pm, notifier := initPaceMaker(t, 3)
-	nve, nveOccurred := pm.ProcessQC(QC(2))
-	assert.True(t, !nveOccurred && nve == nil)
-	notifier.AssertExpectations(t)
-	assert.Equal(t, uint64(3), pm.CurView())
+// TestProcessTC_SkipIncreaseViewThroughTC tests that ActivePaceMaker increases view when receiving TC,
+// if applicable, by skipping views
+func (s *ActivePaceMakerTestSuite) TestProcessTC_SkipIncreaseViewThroughTC() {
+	tc := helper.MakeTC(helper.WithTCView(3),
+		helper.WithTCNewestQC(s.livenessData.NewestQC))
+	expectedLivenessData := &hotstuff.LivenessData{
+		CurrentView: tc.View + 1,
+		LastViewTC:  tc,
+		NewestQC:    tc.NewestQC,
+	}
+	s.persist.On("PutLivenessData", expectedLivenessData).Return(nil).Once()
+	s.notifier.On("OnStartingTimeout", expectedTimerInfo(4, model.ReplicaTimeout)).Return().Once()
+	s.notifier.On("OnTcTriggeredViewChange", tc, uint64(4)).Return().Once()
+	nve, err := s.paceMaker.ProcessTC(tc)
+	require.NoError(s.T(), err)
+	s.notifier.AssertExpectations(s.T())
+	require.Equal(s.T(), uint64(4), s.paceMaker.CurView())
+	require.True(s.T(), nve.View == 4)
+	require.Equal(s.T(), tc, s.paceMaker.LastViewTC())
+
+	tc = helper.MakeTC(helper.WithTCView(12),
+		helper.WithTCNewestQC(s.livenessData.NewestQC),
+		helper.WithTCNewestQC(QC(3)))
```

Comment:
```suggestion
		helper.WithTCNewestQC(s.livenessData.NewestQC),
```
setting `NewestQC` twice